packages/react/src/components/header/components/headerActionBar/HeaderActionBar.tsx

```diff
@@ -9,9 +9,70 @@ import { useCallbackIfDefined, useEnterOrSpacePressCallback } from '../../../../
 import { HeaderActionBarMenuItem } from '../headerActionBarItem';
 import styles from './HeaderActionBar.module.scss';
 import HeaderActionBarLogo from './HeaderActionBarLogo';
+import { useHeaderContext } from '../../HeaderContext';
 
 const classNames = styleBoundClassNames(styles);
 
+enum TabBarrierPosition {
+  top = 'top',
+  bottom = 'bottom',
+}
+
+type TabBarrierProps = {
+  id: string;
+  tabIndex: number;
+  'aria-hidden': boolean;
+};
+
+const defaultBarrierProps: Partial<TabBarrierProps> = {
+  tabIndex: 0,
+  'aria-hidden': true,
+};
+
+const elementIsInViewport = (element: Element) => {
+  const { top, left, bottom, right } = element.getBoundingClientRect();
+  const { innerHeight, innerWidth } = window;
+
+  return top > 0 && left > 0 && bottom < innerHeight && right < innerWidth;
+};
+
+const findFocusableElementsWithin = (element: HTMLElement) => {
+  const focusable = element.querySelectorAll('a, button, textarea, input[type="text"], select');
+
+  return Array.from(focusable).filter((item) => elementIsInViewport(item));
+};
+
+const focusToActionBar = (position: TabBarrierPosition, element?: HTMLElement) => {
+  if (element) {
+    const focusableElements = findFocusableElementsWithin(element);
+
+    if (focusableElements.length) {
+      (focusableElements[
+        position === TabBarrierPosition.top ? 0 : focusableElements.length - 1
+      ] as HTMLElement).focus();
+    }
+  }
+};
+
+const ContentTabBarrier = ({ onFocus }: { onFocus: () => void }): JSX.Element => {
+  /* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */
+  return <div {...defaultBarrierProps} onFocus={onFocus} />;
+};
+
+const addDocumentTabBarrier = (position: TabBarrierPosition, actionBar?: HTMLElement): HTMLDivElement => {
+  const element = document.createElement('div');
+  element.className = 'hds-actionbar-tab-barrier';
+  element.tabIndex = defaultBarrierProps.tabIndex;
+  element['aria-hidden'] = defaultBarrierProps.tabIndex['aria-hidden'];
+  element.addEventListener('focus', () => focusToActionBar(position, actionBar));
```

Comment:
Typo `defaultBarrierProps.tabIndex['aria-hidden']`

Reply:
https://github.com/City-of-Helsinki/helsinki-design-system/pull/1120/commits/d29c00e2229c574192405583d29be724a374714d